src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/stash/StashApiClient.java

```diff
@@ -169,10 +169,23 @@ public StashPullRequestMergeableResponse getPullRequestMergeStatus(String pullRe
     }
   }
 
-  public boolean mergePullRequest(String pullRequestId, String version) throws StashApiException {
+  public String mergePullRequest(String pullRequestId, String version) throws StashApiException {
     String path = pullRequestPath(pullRequestId) + "/merge?version=" + version;
     String response = postRequest(path, null);
-    return !response.equals(Integer.toString(HttpStatus.SC_CONFLICT));
+
+    try {
+      StashPullRequestResponseValue parsedResponse =
+          mapper.readValue(response, StashPullRequestResponseValue.class);
+
+      if ("MERGED".equals(parsedResponse.getState())) {
+        // Successful merge
+        return "MERGED";
+      }
+    } catch (IOException e) {
+    }
```

Comment:
I think we should at least log the exception

Reply:
OK, makes sense. Errors in proper JSON format don't trigger an exception, so we can log those rare cases when the exception occurs.